src/rdkafka_subscription.c

```diff
@@ -171,16 +171,22 @@ rd_kafka_subscription (rd_kafka_t *rk,
 
 rd_kafka_resp_err_t
 rd_kafka_pause_partitions (rd_kafka_t *rk,
-			   rd_kafka_topic_partition_list_t *partitions) {
-	return rd_kafka_toppars_pause_resume(rk, 1, RD_KAFKA_TOPPAR_F_APP_PAUSE,
-					     partitions);
+                           rd_kafka_topic_partition_list_t *partitions) {
+        return rd_kafka_toppars_pause_resume(rk,
+                                             rd_true/*pause*/,
+                                             rd_false/*sync*/,
```

Comment:
this confused me on first glance. I read the sync comment to mean the argument name, not the behavior.

Reply:
I added a new enum type, rd_async_t, with RD_SYNC and RD_ASYNC to make this, and other places in the future, more readable.